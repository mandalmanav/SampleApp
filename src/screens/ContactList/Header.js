import React from 'react'
import { View, StyleSheet, TextInput } from 'react-native'
import { useDispatch, useSelector } from 'react-redux'
import { SET_SEARCH } from '../../store/constants'


export default ({ }) => {
    let { searchTerm } = useSelector(state => state)
    let dispatch = useDispatch()
    return <View style={styles.headerContainer}>

        <TextInput
            testID='search-box'
            style={styles.searchBar}
            value={searchTerm}
            onChangeText={(value)=>{
               dispatch({
                   type:SET_SEARCH,
                   payload:value
               })
            }}
            placeholder="Search"
        />

    </View>
}

const styles = StyleSheet.create({
    iconContainer: {
        flexDirection: 'column',
        alignItems: 'flex-end'
    },
    addIcon: {
        fontSize: 28,
        color: '#4d90fe'
    },
    contactLabel: {
        fontSize: 24,
        fontWeight: 'bold'
    },
    searchBar: {
        backgroundColor: '#f0f0f0',
        fontSize: 14,
        paddingLeft: 4,
        borderTopRightRadius: 4,
        borderTopLeftRadius: 4,
        borderBottomLeftRadius: 4,
        borderBottomRightRadius: 4,
    },
    headerContainer: {
        paddingLeft: 8,
        paddingRight: 8
    }
})