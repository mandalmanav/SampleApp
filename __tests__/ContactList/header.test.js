import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import Header from '../../src/screens/ContactList/Header'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

describe('Test Search Box', () => {
    const initialState = { contacts: [], searchText: '' }
    const mockStore = configureStore()
    let store, wrapper
    store = mockStore(initialState)
    it('Search box is rendered correctly', () => {
        const { getByTestId } = render(<Provider store={store}><Header /></Provider>)
        expect(getByTestId('search-box')).not.toBeNull()
       })
    it('Placeholder text is \'Search\'', () => {
        const { getByPlaceholderText } = render(<Provider store={store}><Header /></Provider>)
        expect(getByPlaceholderText('Search')).not.toBeNull()
    })
    it('Match Snapshot', () => {
        const component = render(<Provider store={store}><Header /></Provider>)
        expect(component).toMatchSnapshot()
    })
})


