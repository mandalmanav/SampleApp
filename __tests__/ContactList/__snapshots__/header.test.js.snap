// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search Box Match Snapshot 1`] = `
<View
  style={
    Object {
      "paddingLeft": 8,
      "paddingRight": 8,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Search"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#f0f0f0",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "fontSize": 14,
        "paddingLeft": 4,
      }
    }
    testID="search-box"
    underlineColorAndroid="transparent"
  />
</View>
`;
